proc/colbuilder.go

```diff
@@ -158,6 +156,32 @@ func NewColumnBuilder(exprs []ast.FieldExpr) (*ColumnBuilder, error) {
 	}, nil
 }
 
+// check if fieldname is "in" one of the fields in fis, or if
+// one of fis is "in" fieldname, where "in" means "equal or is a suffix of".
+func isIn(fieldname string, fis []fieldInfo) bool {
+	// check if fieldname is "in" one of fis
+	splits := strings.Split(fieldname, ".")
+	for i := range splits {
+		sub := strings.Join(splits[:i+1], ".")
+		for _, fi := range fis {
+			if sub == fi.fullname {
+				return true
+			}
+		}
+	}
+	// check if one of fis is "in" fieldname
+	for _, fi := range fis {
+		splits := strings.Split(fi.fullname, ".")
+		for i := range splits {
+			prefix := strings.Join(splits[:i+1], ".")
+			if prefix == fieldname {
+				return true
+			}
+		}
+	}
```

Comment:
I think the prefixes you are deriving here should be equivalent to all the keys in `seenRecords`?  I don't have a strong preference for changing this, I think its much clearer the way you've written it here and the cost of the string operations here should be modest, but wanted to point it out...

Reply:
Hmm, as I was writing this I was trying to prioritize clarity rather than performance optimization given this is compile time . I'll confess I hadn't noticed the overlap with `seenRecords`...